### PR TITLE
Create TimeMap LDPCv for binary resource.

### PR DESCRIPTION
**Pull request by lsitu**
Fixes https://jira.duraspace.org/browse/FCREPO-2644
Create TimeMap LDPCv for binary resource.

# What does this Pull Request do?
Create TimeMap under the parent container of the binary resource with the name of the binary node.
Refactored codes to move the signature getTimeMap() to FedoraResource to make it easier to access the TimeMap for FedoraResource/FedoraBinary.

### CODE REVIEW

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java [hunk 1]
```diff
@@ -445,6 +448,44 @@ private static void decorateContentNode(final Node contentNode, final Collection
...
+    }
+
+    @Override
+    public FedoraResource findOrCreateTimeMap() {
+        final Node ldpcvParentNode;
+        final Node ldpcvNode;
+
+        try {
+            final Node container = getContainingNode(getNode()).get();
+            ldpcvParentNode = findOrCreateChild(container, LDPCV_TIME_MAP, NT_FOLDER);
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java]: As a comment to this method, could you please detail the design of the TimeMap containers?

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java [hunk 2]
```diff
@@ -445,6 +448,44 @@ private static void decorateContentNode(final Node contentNode, final Collection
     }
 
     @Override
+    public FedoraResource getTimeMap() {
+        try {
+            final Node container = getContainingNode(getNode()).get();
```
Comment (mohideen) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java]: @lsitu Is this implementation assuming that the BinaryServiceImpl will be refactored to use a new nt:folder node as an enclosing node for the binary nt:file node.

https://github.com/lsitu/fcrepo4/blob/60d6c54bcb8ed28563fe9d8ad86814b6c69a4dc0/fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/BinaryServiceImpl.java#L57

Comment (lsitu) [reply fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java]: @mohideen No. I don't think so. This is retrieve the TimeMap for binary.

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java (hunk 1 above)
Comment (lsitu) [reply fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java]: As we discussed during the sprint wrap-up meeting this morning, the TimeMap for binary will be created inside the TimeMap of the parent container that holds the binary resource, which has the same name as the binary node.

**Comment by awoods** [comment]
Superseded by: https://github.com/fcrepo4/fcrepo4/pull/1258

**Pull request closed without merging**